internal/engine/filter/filter.go

```diff
@@ -45,9 +50,16 @@ type Filter struct {
 	Criteria Criteria
 }
 
-func (f Filter) String() string {
+// DebugString returns text representation of the filter. If extraInfo is
+// true, debugging information like Gmail search representation and URL are
+// included.
+func (f Filter) DebugString(extraInfo bool) string {
```

Comment:
I would say, let's keep `DebugString` without parameters and have some duplication. It doesn't hurt and makes the signature more understandable.